pkg/threescale/backend/backend.go

```diff
@@ -245,11 +244,10 @@ func (b *Backend) localReport(cacheKey string, metrics api.Metrics) {
 	defer application.Unlock()
 
 	for metric, incrementBy := range metrics {
-		cachedValue, ok := application.LimitCounter[metric]
-		if ok {
-			for _, limit := range cachedValue {
-				limit.current += incrementBy
-			}
+		cachedValue, ok := application.LocalState[metric]
+		if ok && len(cachedValue) > 0 {
+			// for local reporting, it is only necessary for us to update the lowest known granularity
```

Comment:
This changes the previous behaviour, doesn't it? I think we chatted about this, and my initial thought was wrong (and I think this is doing that, and so wrong as well :). When we report locally, we want to keep the current value for all limited granularities so that after we unlock the next incoming request authorising against this metric would see the increments in each granularity, that is, if it was limited in minutes and in days, perhaps the minute elapses and becomes 0, but the days would keep counting and we'd want the incremented value in the day period to check the daily limit.

Reply:
hmm, no I don't think that's what we discussed. For local reporting, we said we only needed to update the lowest granularity. In cases where we had a successful report and a failed authorisation (ie a sequence that would update our local representation of the remote state), then we would add to each granularity, my understanding was that we were treating these cases separately.  See https://github.com/3scale/3scale-istio-adapter/pull/121/files#diff-d1d4808f2b00377237f629f7583be175R463 for this behaviour

Reply:
Leaving a note here based on our discussion over slack: Decided it was imperative that we update for all granularities for rate limiting purposes. So local reporting and authz always needs to iterate on the full slice of known usage reports.

Fixing